rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_publisher.hpp

```diff
@@ -97,32 +81,13 @@ class LifecyclePublisher : public LifecyclePublisherInterface,
   virtual void
   publish(const MessageT & msg)
   {
-    if (!enabled_) {
+    if (!this->is_activated()) {
       log_publisher_not_enabled();
       return;
     }
     rclcpp::Publisher<MessageT, Alloc>::publish(msg);
   }
 
-  virtual void
-  on_activate()
-  {
-    enabled_ = true;
-  }
-
-  virtual void
-  on_deactivate()
-  {
-    enabled_ = false;
-    should_log_ = true;
```

Comment:
I think this breaks the previous logging behavior; we no longer set `should_log_` to true after a deactivation. TBH, I'm not sure if this logging behavior is necessary though; I guess it could be useful for debugging, but it also seems like it could be annoying in some scenarios :shrug:

Reply:
Fixed in https://github.com/ros2/rclcpp/pull/1863/commits/469f5d74b09a6dd2e73c733d323252730d89ca4e